src/clj/puppetlabs/services/jruby_pool_manager/impl/jruby_internal.clj

```diff
@@ -143,31 +143,57 @@
     (log/infof (i18n/trs "Cleaned up old JRubyInstance with id {0}."
                          (:id instance)))))
 
+(schema/defn ^:always-validate
+  initial-borrows-value :- (schema/maybe schema/Int)
+  [id :- schema/Int
+   total-instances :- schema/Int
+   total-borrows :- schema/Int
+   initial-jruby? :- schema/Bool]
+  (when initial-jruby?
+    (let [which-step (inc (mod id total-instances))
+          step-size (quot total-borrows total-instances)]
+      ;; If total-instances is larger than total-borrows then step-size will be 0
+      (if (= 0 step-size)
```

Comment:
`if` => `when-not` and lose the nil branch?

Reply:
I always forget the `not` variations.